### ISSUE TITLE
TieredMenu: Submenu 'popup' behavior different than PrimeVue

### ISSUE THREAD

**Issue by msalihaltun** [primefaces/primereact#5975]
### Describe the bug

I have two forked demos that shows the difference in behavior. In `10.2` TieredMenu submenus were activated by hover. In the latest version they can only be activated by clicking.

 - Latest version

https://stackblitz.com/edit/zjjwcj?file=src%2FApp.jsx 

 - v10.2

 https://stackblitz.com/edit/ompses?file=src%2FApp.jsx

### Reproducer

https://stackblitz.com/edit/ompses?file=src%2FApp.jsx

### PrimeReact version

latest

### React version

18.x

### Language

TypeScript

### Build / Runtime

Vite

### Browser(s)

Tested in chrome

### Steps to reproduce the behavior

Use the demo to see the different behavior.

### Expected behavior

I expect the submenus to activate by hover.

**Comment by melloware** [comment]
@msalihaltun it appears PrimeVue is the same way: https://primevue.org/tieredmenu/

It was done as part of ARIA accessibility and keyboard support: https://primereact.org/tieredmenu/#accessibility

See #5657

**Comment by msalihaltun** [comment]
Thank you for your response.

**Comment by kyybo** [comment]
> @msalihaltun it appears PrimeVue is the same way: https://primevue.org/tieredmenu/
> 
> It was done as part of ARIA accessibility and keyboard support: https://primereact.org/tieredmenu/#accessibility
> 
> See #5657

@melloware  @msalihaltun 
Hi, 
It seems to me that there's a difference in behavior between PrimeVue and PrimeReact.
A first click is required to open a first submenu, then once the TieredMenu has the focus, you can navigate inside it with the hover.
This behavior is the same for PrimeVue and PrimeReact.

However, if you activate the 'popup' property, the behavior described above no longer works in PrimeReact, as all submenus (and sub-submenus) can only be activated by clicking. (https://primereact.org/tieredmenu/#popup)
In PrimeVue, with the 'popup' property, once the first sub-menu has been opened, you can navigate through the TieredMenu using hovers (https://primevue.org/tieredmenu/#popup).

**Comment by kyybo** [comment]
Hello there !
If anyone is looking for a simple solution, waiting for a PrimeReact fix, this CSS should do the work :
`#your_tieredmenu_id li:hover>ul {
    display: block !important;
    top: 0px;
    left: 100%;
}
`

**Comment by leevva** [comment]
This is still relevant. I would like to avoid unnecessary clicks.
Submenu does not show on hover in popup mode. 
Workaround does not help, because the submenu is displayed outside the visible area (on the right).

**Comment by kyybo** [comment]
> This is still relevant. I would like to avoid unnecessary clicks. Submenu does not show on hover in popup mode. Workaround does not help, because the submenu is displayed outside the visible area (on the right).

Indeed, if your button is on the right edge of the screen, we prefer to display the sub-menu on the left. In this case you need to apply `left: -100%` to my workaround.
In the PrimeReact code, you can [find here](https://github.com/primefaces/primereact/blob/1614d6a50a01921d3f3b75956dc060ef2bbbf9aa/components/lib/tieredmenu/TieredMenuSub.js#L38) under which condition they give the CSS class `p-submenu-list-flipped` to the TieredMenu that applies the `left: -100%`.
I hope this helps.

**Comment by leevva** [comment]
@kyybo Thank you, it works.
